### ISSUE TITLE
Documentation overhaul

### ISSUE THREAD

**Issue by JeS24** [einsteinpy/einsteinpy#630]
### 🐞**Problem**
(Moved from #614)

We need help with the following with respect to the API documentation ("API reference") of the various EinsteinPy submodules:
1. Fixing grammatical errors and making the wording simpler at certain places.
2. Expanding the current documentation, something like what [`sunpy`](https://docs.sunpy.org/en/stable/code_ref/index.html) or [`astropy`](https://docs.astropy.org/en/stable/constants/index.html) API references read like.
3. Adding new example Jupyter notebooks that serve as tutorials or improving old ones. (See also #626). 

Some good starting points are perhaps the docs for the following submodules:
* [`coordinates`](https://docs.einsteinpy.org/en/latest/api/coordinates/coordinates_index.html)
* [`integrators`](https://docs.einsteinpy.org/en/latest/api/integrators/integrators_index.html)
* [`metric`](https://docs.einsteinpy.org/en/latest/api/metric/metric_index.html)
* [`symbolic`](https://docs.einsteinpy.org/en/latest/api/symbolic/symbolic_index.html)
* [`geodesic`](https://docs.einsteinpy.org/en/latest/api/geodesic/geodesic.html)

### ❕ **Relevant information**

* The source for the EinsteinPy docs can be found here: https://github.com/einsteinpy/einsteinpy/tree/main/docs/source/api, while the compiled docs are available at https://docs.einsteinpy.org/en/latest/api/index.html.
*  If you want to know what the code does, a good place to start would be the [`source/examples/`](https://github.com/einsteinpy/einsteinpy/tree/main/docs/source/examples) folder, since it contains tutorials in the form of Jupyter notebooks. You can run the code there and understand what it is doing. You can then use that knowledge to improve the API docs (at least for the parts of the API present in the tutorials).
* We use reStructuredText (reST) for writing the docs in the `numpydoc` style. You can find a quick primer on reST here: https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html, while an example of the `numpydoc` style is available here: https://numpydoc.readthedocs.io/en/latest/example.html.

 ### 📋**Steps to solve the problem**

Please consult the [developer docs](https://docs.einsteinpy.org/en/latest/dev_guide.html) to setup your dev environment and get started. Also, feel free to communicate, if you hit a snag. You can do so in our [Gitter](https://gitter.im/EinsteinPy-Project/EinsteinPy) or [Matrix/Element](https://matrix.to/#/#einsteinpy:matrix.org) chat rooms (recommended), or you can also reply here.

 * Comment below about what you've started working on.
 * Add, commit, push your changes
 * Submit a pull request.
 * Add this in the comments:
 	- `Addresses #<put issue number here>` if you are partially fixing the issue.
 	- `Fixes #<put issue number here>` if you are completely fixing the issue.
 * Ask for a review in comments section of pull request
 * Celebrate your contribution to this project 🎉

**Comment by Kennshen** [comment]
Hi, I am going to start working on the API documentation for the metrics submodule.

**Comment by JeS24** [comment]
Hi @Kennshen Sure. Please go ahead.

**Comment by Kennshen** [comment]
@JeS24 I have been trying to open EinsteinPy in editable mode, but when I follow the guide, I keep getting the following error message "./einsteinpy/[dev] is not a valid editable requirement. It should either be a path to a local project or a VCS URL (beginning with bzr+http, bzr+https, bzr+ssh, bzr+sftp, bzr+ftp, bzr+lp, bzr+file, git+http, git+https, git+ssh, git+git, git+file, hg+file, hg+http, hg+https, hg+ssh, hg+static-http, svn+ssh, svn+http, svn+https, svn+svn, svn+file)." Is there something I am missing?

**Comment by JeS24** [comment]
@Kennshen Can you share the command that led to that error, i.e. the command that you were using to install EPy in editable mode? It should be `pip install -e ./einsteinpy/[dev]`. The `-e` flag denotes editable and the `[dev]` marker signals pip to also install the developer dependencies. If the same error comes up, can you try the command without the `[dev]` marker?

**Comment by Kennshen** [comment]
I did try using the above method. I have also tried using the command without the [dev] marker and still got the same error

**Comment by JeS24** [comment]
@Kennshen I think, the path that you are giving to `pip install -e` is incorrect. Can you check if the folder structure looks like this, when you try to install it?
```
current_folder
└───einsteinpy
```
Where `einsteinpy` was created previously by `git clone` command, as in the guide. You need to be in `current_folder` and then type in the command, `pip install -e ./einsteinpy/[dev]`, and it will work. Otherwise, navigate (`cd`) to the cloned `einsteinpy` folder and try `pip install -e ./[dev]`.

**Comment by dishb** [comment]
I would suggest making a list of all modules that need to be documented. The list would contain whether they are open or not, competition status, and who is currently working on it. 

This would also help people find modules that need documentation that are being worked on but not completed. More collaboration.

**Comment by JeS24** [comment]
@dishb That's a good suggestion. While the body of this issue has some of the modules listed and linked, there is no status tracking. Let me see, what I can do here; maybe leverage GitHub Projects.

**Comment by dishb** [comment]
@JeS24  - Github Projects sounds like a great idea! 😄

**Comment by Snerdify** [comment]
Hey has this issue been assigned or can I take it up?

**Comment by dishb** [comment]
There are many parts to this issue. I don't think one person can accomplish it.

**Comment by JeS24** [comment]
@Snerdify Feel free to contribute. As @dishb said, it is a large set of issues. However, for each individual portion that is overhauled, we will create different issues as a way of tracking progress. Meanwhile, let me see if I can set up GitHub projects. I might not have the right permissions on the repo.

**Comment by dishb** [comment]
@JeS24 I believe only maintainers with write access can create Projects. (Or collaborators)

**Comment by JeS24** [comment]
> @JeS24 I believe only maintainers with write access can create Projects. (Or collaborators)

Yeah, in terms of [GitHub roles](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/repository-roles-for-an-organization), only maintainers can create Projects, it seems. Either way, for now, we can use separate issues depending on PRs to status-track.

**Comment by JeS24** [comment]
@dishb @Snerdify @Kennshen I have added this issue to a GitHub project [here](https://github.com/orgs/einsteinpy/projects/1). If you are working on any of the components of this issue, please tell me. I'll create a corresponding issue and add you as an assignee to it on the project board. It will serve as the status tracker.

**Comment by JeS24** [comment]
Also, in case you cannot see the board publicly, give it some time. It will be up soon.

**Comment by GaganaMD** [comment]
Hey! I would like to be assigned to this issue.

**Comment by JeS24** [comment]
@GaganaMD Would you like to work on this issue or #650? Although, if you are working on #650, you can also improve the docs in `coordinates` simultaneously. At this time, updating docs for `coordinates` doesn't make sense, till either #644 or a similar PR is merged. But if you want to work on the docs for some other module in the list above, please mention it here and feel free to work on it.

**Comment by GaganaMD** [comment]
Oh. So can I start working on issue #650 first then, or should I wait?

**Comment by JeS24** [comment]
@GaganaMD You can start working on #650.

**Comment by TiPa7** [comment]
Hey, I'd like to help. Is there any special issue you would want to be fixed most urgently in the Docs?
